### PR TITLE
Add a dedicated fixer to fixup code in expression trees that is now calling restricted apis

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/74030

This was a request from the LDM for users who run into cases where their expression trees no longer compile because the compiler now picks an overload of the method they were calling that is not legal to use in an expression tree.  For example, switching to a `params ReadOnlySpan<...>` method from a `params T[]` method.

The fixer switches the code to use explicit types to ensure this continues working.

### CODE REVIEW

File: src/Analyzers/CSharp/CodeFixes/CSharpCodeFixes.projitems [hunk 1]
```diff
@@ -10,7 +10,9 @@
   </PropertyGroup>
   <ItemGroup>
     <EmbeddedResource Include="$(MSBuildThisFileDirectory)CSharpCodeFixesResources.resx" GenerateSource="true" Link="CSharpCodeFixesResources.resx" />
-    <None Include="$(MSBuildThisFileDirectory)CSharpCodeFixesResources.resx" />
+    <None Include="$(MSBuildThisFileDirectory)CSharpCodeFixesResources.resx">
+      <SubType>Designer</SubType>
+    </None>
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/CodeFixes/CSharpCodeFixes.projitems:15]:
```suggestion
    <None Include="$(MSBuildThisFileDirectory)CSharpCodeFixesResources.resx" />
```

File: src/Analyzers/CSharp/CodeFixes/UseExplicitArrayInExpressionTree/CSharpUseExplicitArrayInExpressionTreeCodeFixProvider.cs [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                    RegisterCodeFix(
+                        context, CSharpCodeFixesResources.Use_explicit_array, nameof(CSharpCodeFixesResources.Use_explicit_array));
+                }
+            }
+        }
+    }
+
+    protected override async Task FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, SyntaxEditor editor, CancellationToken cancellationToken)
+    {
+        var invocationExpressions = diagnostics
```
Comment (ToddGrun) [inline src/Analyzers/CSharp/CodeFixes/UseExplicitArrayInExpressionTree/CSharpUseExplicitArrayInExpressionTreeCodeFixProvider.cs:109]: >invocationExpressions [](http://example.com/codeflow?start=12&length=21)

is this used?

File: src/Analyzers/CSharp/CodeFixes/UseExplicitArrayInExpressionTree/CSharpUseExplicitArrayInExpressionTreeCodeFixProvider.cs [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                    continue;
+
+                for (int i = 0, n = currentMethod.Parameters.Length - 1; i < n; i++)
+                {
+                    var originalParameter = originalMethod.Parameters[i];
+                    var currentParameter = currentMethod.Parameters[i];
+
+                    // Different parameter types, look for another sibling that matches
+                    if (!originalParameter.Type.Equals(currentParameter.Type))
+                        continue;
```
Comment (ToddGrun) [inline src/Analyzers/CSharp/CodeFixes/UseExplicitArrayInExpressionTree/CSharpUseExplicitArrayInExpressionTreeCodeFixProvider.cs:65]: >continue; [](http://example.com/codeflow?start=24&length=9)

is this intending to continue on the outer loop?

Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/CodeFixes/UseExplicitArrayInExpressionTree/CSharpUseExplicitArrayInExpressionTreeCodeFixProvider.cs:65]: yup.  will fix.

File: src/Analyzers/CSharp/CodeFixes/UseExplicitArrayInExpressionTree/CSharpUseExplicitArrayInExpressionTreeCodeFixProvider.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/CodeFixes/UseExplicitArrayInExpressionTree/CSharpUseExplicitArrayInExpressionTreeCodeFixProvider.cs]: Nope.  Removed.

File: src/Analyzers/CSharp/CodeFixes/UseExplicitArrayInExpressionTree/CSharpUseExplicitArrayInExpressionTreeCodeFixProvider.cs [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        return null;
+
+        static bool ParameterTypesMatch(IMethodSymbol originalMethod, IMethodSymbol currentMethod)
+        {
+            for (int i = 0, n = currentMethod.Parameters.Length - 1; i < n; i++)
+            {
+                var originalParameter = originalMethod.Parameters[i];
+                var currentParameter = currentMethod.Parameters[i];
+
+                // Different parameter types, look for another sibling that matches
```
Comment (ToddGrun) [inline src/Analyzers/CSharp/CodeFixes/UseExplicitArrayInExpressionTree/CSharpUseExplicitArrayInExpressionTreeCodeFixProvider.cs:85]: >// Different parameter types, look for another sibling that matches [](http://example.com/codeflow?start=16&length=67)

nit: consider moving to caller

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**